utils/pattern/Cargo.toml

```diff
@@ -0,0 +1,30 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+[package]
+name = "icu_pattern"
+description = "ICU pattern utilities"
+version = "0.1.0"
+authors = ["The ICU4X Project Developers"]
+edition = "2018"
+readme = "README.md"
+repository = "https://github.com/unicode-org/icu4x"
+license-file = "../../LICENSE"
+include = [
+    "src/**/*",
+    "examples/**/*",
+    "benches/**/*",
+    "Cargo.toml",
+    "README.md"
+]
+
+[dependencies]
+thiserror = "1"
+
+[dev-dependencies]
+iai = "0.1"
```

Comment:
In this specific case, I think iai brings a lot of value to the table. I also don't have reason to believe this means over-optimizing for a particular platform, because we only look at relative deltas, and a delta on one platform is likely correlated with a delta on another.

And in general, I have no problem introducing platform-specific build tools, especially if built for an open platform that anyone can install for free.